src/components/switchComponent/index.tsx

```diff
@@ -1,17 +1,49 @@
-import React, { FC } from 'react';
+import React, { FC, ReactNode } from 'react';
+import { FlexboxProps } from '@chakra-ui/styled-system';
 import {
   Switch as ChakraSwitch,
   SwitchProps as ChakraSwitchProps,
+  FormControl,
+  FormLabel,
 } from '@chakra-ui/react';
 
 export type SwitchProps = Pick<
   ChakraSwitchProps,
-  'onChange' | 'isChecked' | 'variant'
->;
+  'onChange' | 'isChecked' | 'variant' | 'isDisabled'
+> & {
+  label?: ReactNode;
+  labelPosition?: FlexboxProps['flexDirection'];
+};
 
-const Switch: FC<SwitchProps> = ({ onChange, isChecked, variant }) => {
+const Switch: FC<SwitchProps> = ({
+  onChange,
+  isChecked,
+  variant,
+  isDisabled,
+  label,
+  labelPosition = 'row',
+}) => {
   return (
-    <ChakraSwitch onChange={onChange} isChecked={isChecked} variant={variant} />
+    <FormControl
+      width="min-content"
+      display="flex"
+      alignItems="center"
+      gap="sm"
+      flexDirection={labelPosition}
+    >
+      {label ? (
+        <FormLabel htmlFor="switch-component" mb="0">
+          {label}
+        </FormLabel>
+      ) : null}
+      <ChakraSwitch
+        id="switch-component"
+        onChange={onChange}
+        isChecked={isChecked}
+        variant={variant}
+        isDisabled={isDisabled}
```

Comment:
you still have pointer cursor on the label even if it's disabled. could you check this please?